### ISSUE TITLE
Using SkipTestException causes error in JUnitTestRunnerListener

### ISSUE THREAD

**Issue by eschrock** [kotest/kotest#1034]
I have a set of tests that may or may not run depending on the environmental configuration, so I was excited to make use of SkipTestException. Unfortunately, no matter what I do I always get an exception like this in my test output (also displayed with "--info" to gradle):

```
    2019-10-02 16:39:28.964 [kotlintest-engine-0] ERROR i.k.r.junit5.JUnitTestRunnerListener - Error in JUnit Platform listener
    java.lang.RuntimeException: A skipped test should never have been started
        at io.kotlintest.runner.junit5.JUnitTestRunnerListener.afterSpecClass(JUnitTestRunnerListener.kt:189)
        at io.kotlintest.runner.jvm.IsolationTestEngineListener.afterSpecClass(IsolationTestEngineListener.kt:106)
        at io.kotlintest.runner.jvm.spec.SpecExecutor$execute$$inlined$invoke$lambda$1.invoke(SpecExecutor.kt:68)
        at io.kotlintest.runner.jvm.spec.SpecExecutor$execute$$inlined$invoke$lambda$1.invoke(SpecExecutor.kt:20)
        at io.kotlintest.runner.jvm.spec.SpecExecutor.withExecutor(SpecExecutor.kt:31)
        at io.kotlintest.runner.jvm.spec.SpecExecutor.execute(SpecExecutor.kt:37)
        at io.kotlintest.runner.jvm.TestEngine$submitSpec$1.run(TestEngine.kt:111)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:835)
```

I am running 3.4.2 with the junit5 runner through gradle. While the tests show as SKIPPED, the above exception prevents any subsequent test suites from running. So instead of seeing several hundred tests with some subset skipped, I see only the first suite with skipped tests and then nothing else (they don't even run). I can see this even with a very simple test case:

```
class MyTest : StringSpec() {
    init {
        "skip this test" {
            throw SkipTestException("skip it")
        }
    }
}
```

**Comment by LeoColman** [comment]
Ok, it seems that this bug was introduced in a commit made by @sksamuel in 29/06/2019:

```
class JUnitTestRunnerListener

...

// Line 189
if (started.contains(it.first))
                    throw RuntimeException("A skipped test should never have been started")

```

Sam, is there a reason for this to be here? Or can we remove this check?
If there isn't, we should remove it and improve our test suite to get this case, as we're mocking the TestListener in `SkipTestException` test.

**Comment by sksamuel** [comment]
It's a sanity check to ensure we're compliment with the junit spec. If a test is marked as started, we can't then mark it as ignored later.

Usage of skip exception might need to be re-done, so that we never mark a test as started until after the test has completed, so we can check for this exception. It can be done alongside the other issue causing passed tests to appear ignored when there are other failures.

I'll fix this for 3.4.3

**Comment by sksamuel** [comment]
JUnit Platform doesn't allow a test to be skipped after it has been started.
Therefore this functionality will be removed in 4.0 and a function "isActive" will be added to test config to allow you to use a function to determine if a test should run.
